source/extensions/filters/http/oauth2/oauth_client.cc

```diff
@@ -89,14 +91,24 @@ void OAuth2ClientImpl::onSuccess(const Http::AsyncClient::Request&,
 
   // TODO(snowp): Should this be a pgv validation instead? A more readable log
   // message might be good enough reason to do this manually?
-  if (!response.has_access_token() || !response.has_expires_in()) {
-    ENVOY_LOG(debug, "No access token or expiration after asyncGetAccessToken");
+  if (!response.has_access_token()) {
+    ENVOY_LOG(debug, "No access token after asyncGetAccessToken");
     parent_->sendUnauthorizedResponse();
     return;
   }
 
   const std::string access_token{PROTOBUF_GET_WRAPPED_REQUIRED(response, access_token)};
-  const std::chrono::seconds expires_in{PROTOBUF_GET_WRAPPED_REQUIRED(response, expires_in)};
+
+  std::chrono::seconds expires_in = default_expires_in_;
+  if (response.has_expires_in()) {
+    expires_in = std::chrono::seconds{response.expires_in().value()};
+  }
+  if (expires_in <= 0s) {
```

Comment:
This inherently adds a check that the response's `expires_in` field doesn't contain some logically invalid negative duration. Probably a good thing to send a local reply instead of setting client cookies etc.